xls/dslx/interpreter_main.cc

```diff
@@ -169,16 +173,39 @@ absl::StatusOr<TestResult> RealMain(
   XLS_ASSIGN_OR_RETURN(std::string module_name, PathToName(entry_module_path));
 
   std::unique_ptr<AbstractRunComparator> run_comparator;
-  switch (compare_flag) {
-    case CompareFlag::kNone:
-      break;
-    case CompareFlag::kJit:
+  // When comparison is on, use the same mode (JIT/interpreter) as the
+  // comparator. Otherwise, use the same mode as the evaluator flag. Using the
+  // IR interpreter requires --run_quickcheck_when_interpreting.
```

Comment:
q: is that because it's not a common usecase that we want to not run the quickcheck by default when the evaluator is the interpreter?

one alternative could be to have a `--run_quickcheck` flag w/ different default depending on the evaluator being set (but maybe that would be more confusing?)